app/services/credit_notes/create_service.rb

```diff
@@ -226,5 +230,17 @@ def adjust_credit_note_tax_rounding
     def all_rounding_tax_adjustments
       credit_note.invoice.credit_notes.sum(&:taxes_rounding_adjustment)
     end
+
+    def associated_wallet
+      @associated_wallet ||= invoice.associated_active_wallet
+    end
+
+    def voiding_credits
+      return 0 unless invoice.credit? && associated_wallet.present?
+
+      # wallet transactions don't have cents amount, so we have to convert value into full amount
+      # and then convert money into amount of credits
+      credit_note.refund_amount_cents / 100 / associated_wallet.rate_amount
```

Comment:
I guess that you have to take care of the currency for this conversion (currency without subunits or with more decimals than EUR/USD...):

```suggestion
      credit_note.refund_amount_cents / credit_note.refund_amount.currency.subunit_to_unit / associated_wallet.rate_amount
```

Reply:
ah, nice!
Thank you!